addOns/client/src/test/java/org/zaproxy/addon/client/automation/ClientSpiderJobUnitTest.java

```diff
@@ -0,0 +1,247 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.withSettings;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import org.apache.commons.httpclient.URIException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.quality.Strictness;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.model.Model;
+import org.yaml.snakeyaml.Yaml;
+import org.zaproxy.addon.automation.AutomationEnvironment;
+import org.zaproxy.addon.automation.AutomationJob.Order;
+import org.zaproxy.addon.automation.AutomationProgress;
+import org.zaproxy.addon.automation.ContextWrapper;
+import org.zaproxy.addon.client.ClientOptions;
+import org.zaproxy.addon.client.ExtensionClientIntegration;
+import org.zaproxy.addon.client.spider.ClientSpider;
+import org.zaproxy.addon.commonlib.Constants;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.testutils.TestUtils;
+import org.zaproxy.zap.utils.I18N;
+
+public class ClientSpiderJobUnitTest extends TestUtils {
+
+    private ExtensionLoader extensionLoader;
+    private ExtensionClientIntegration extClient;
+
+    @BeforeEach
+    void setUp() {
+        mockMessages(new ExtensionClientIntegration());
+
+        extensionLoader =
+                mock(ExtensionLoader.class, withSettings().strictness(Strictness.LENIENT));
+        extClient = mock(ExtensionClientIntegration.class);
+        given(extensionLoader.getExtension(ExtensionClientIntegration.class)).willReturn(extClient);
+
+        Control.initSingletonForTesting(Model.getSingleton(), extensionLoader);
+    }
+
+    @Test
+    void shouldReturnDefaultFields() {
+        // Given / When
+        ClientSpiderJob job = new ClientSpiderJob();
+
+        // Then
+        assertThat(job.getType(), is(equalTo("spiderClient")));
+        assertThat(job.getName(), is(equalTo("spiderClient")));
+        assertThat(job.getOrder(), is(equalTo(Order.LAST_EXPLORE)));
+        assertThat(job.getParamMethodObject(), is(nullValue()));
+        assertThat(job.getParamMethodName(), is(nullValue()));
+        assertValidTemplate(job.getTemplateDataMin());
+        assertValidTemplate(job.getTemplateDataMax());
+    }
+
+    @Test
+    void shouldVerifyWithoutParameters() {
+        // Given
+        AutomationProgress progress = new AutomationProgress();
+        ClientSpiderJob job = new ClientSpiderJob();
+        job.setJobData(null);
+
+        // When
+        job.verifyParameters(progress);
+
+        // Then
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.getErrors(), is(empty()));
```

Comment:
Also check that warnings is empty?

Reply:
Cant see this point in that, will remove the getErrors check..